code/game/mecha/mecha_wreckage.dm

```diff
@@ -29,97 +38,87 @@
-			return
+		QDEL_NULL(AI)
+	QDEL_LIST(crowbar_salvage)
+	return ..()
 
-		if(!I.use_tool(src, user, 0, volume=50))
-			return
-
-		var/type = prob(70) ? pick(welder_salvage) : null
-		if(type)
-			var/N = new type(get_turf(user))
-			user.visible_message("[user] cuts [N] from [src].", "<span class='notice'>You cut [N] from [src].</span>")
-			if(istype(N, /obj/item/mecha_parts/part))
-				welder_salvage -= type
-			salvage_num--
-		else
-			to_chat(user, "<span class='warning'>You fail to salvage anything valuable from [src]!</span>")
+/obj/structure/mecha_wreckage/examine(mob/user)
+	. = ..()
+	if(!AI)
 		return
+	. += "<span class='notice'>The AI recovery beacon is active.</span>"
 
-	else if(I.tool_behaviour == TOOL_WIRECUTTER)
-		if(salvage_num <= 0)
-			to_chat(user, "<span class='warning'>You don't see anything that can be cut with [I]!</span>")
-			return
-		else if(wirecutters_salvage && wirecutters_salvage.len)
-			var/type = prob(70) ? pick(wirecutters_salvage) : null
-			if(type)
-				var/N = new type(get_turf(user))
-				user.visible_message("[user] cuts [N] from [src].", "<span class='notice'>You cut [N] from [src].</span>")
-				salvage_num--
-			else
-				to_chat(user, "<span class='warning'>You fail to salvage anything valuable from [src]!</span>")
-
-	else if(I.tool_behaviour == TOOL_CROWBAR)
-		if(crowbar_salvage && crowbar_salvage.len)
-			var/obj/S = pick(crowbar_salvage)
-			if(S)
-				S.forceMove(user.drop_location())
-				crowbar_salvage -= S
-				user.visible_message("[user] pries [S] from [src].", "<span class='notice'>You pry [S] from [src].</span>")
-			return
-		else
-			to_chat(user, "<span class='warning'>You don't see anything that can be pried with [I]!</span>")
-
+/obj/structure/mecha_wreckage/welder_act(mob/living/user, obj/item/I)
+	..()
+	. = TRUE
+	if(salvage_num <= 0 || !length(welder_salvage))
+		to_chat(user, "<span class='notice'>You don't see anything that can be cut with [I]!</span>")
+		return
+	if(!I.use_tool(src, user, 0, volume=50))
+		return
+	if(prob(30))
+		to_chat(user, "<span class='notice'>You fail to salvage anything valuable from [src]!</span>")
+		return
+	var/type = pick(welder_salvage)
+	var/N = new type(get_turf(user))
+	user.visible_message("[user] cuts [N] from [src].", "<span class='notice'>You cut [N] from [src].</span>")
+	if(!istype(N, /obj/item/stack))
+		welder_salvage -= type
+	salvage_num--
+
+/obj/structure/mecha_wreckage/wirecutter_act(mob/living/user, obj/item/I)
+	..()
+	. = TRUE
+	if(wires_removed)
+		to_chat(user, "<span class='notice'>You don't see anything that can be cut with [I]!</span>")
+		return
+	var/N = new /obj/item/stack/cable_coil(get_turf(user), rand(1,3))
+	user.visible_message("[user] cuts [N] from [src].", "<span class='notice'>You cut [N] from [src].</span>")
+	wires_removed = TRUE
+
+/obj/structure/mecha_wreckage/crowbar_act(mob/living/user, obj/item/I)
+	..()
+	. = TRUE
+	if(crowbar_salvage.len)
+		var/obj/S = pick(crowbar_salvage)

```

Comment:
This needs to remove it from the crowbar_salvage list doesn't it ?

Reply:
yeah forgot that. Fixed now.